File: src/ProjectR/content/scripts/afflictions/Passives/Counter.lua
function GetName()
  return "Counter"
end

function GetType()
  return Passive
end

function GetHookPoints()
  return { AttackDodged, AttackBlocked }
end

function OnAttachment(char)
end

function OnRemoval(char)
end

function Trigger(hookPoint, char)
  if hookPoint == AttackDodged or hookPoint == AttackBlocked then
    local attacker = GetCurrentAttacker()
    attacker:TakeTrueDamage(param1 * 0.2)
  end
end

File: src/ProjectR.Scripting/content/scripts/afflictions/Passives/Counter.cs
using ProjectR.Interfaces.Model;

namespace ProjectR.Scripting.Afflictions
{
    public class Counter : Affliction
    {
        public override string Name { get { return "Counter"; } }

        public override AfflictionType Type { get { return AfflictionType.Passive; } }

        protected override HookPoint[] HookPoints
        {
            get
            {
                return new[]
                {
                    HookPoint.AttackDodged, HookPoint.AttackBlocked, 
                };
            }
        }

        protected override void OnAttackDodged(ICharacter character, double damage)
        {
            CounterAttack(damage);
        }

        protected override void OnAttackBlocked(ICharacter character, ref double damage, ref double reduction)
        {
            CounterAttack(damage);
        }

        private void CounterAttack(double damage)
        {
            GetCurrentAttacker().TakeTrueDamage(damage * .2);
        }
    }
}